### ISSUE TITLE
Annotating classes that augment the prototype do not work correct on exports

### ISSUE THREAD

**Issue by richardspence** [Microsoft/TypeScript#24420]
**Search Terms:** annotation, exports, typings

Annotating exports directly does not return the mutated object, instead it returns the original class.

Given:
```ts

export type IConnect<T> = T & {
    WrappedComponent: T;
} ;

export function connect<T extends Function>(constructor: T): IConnect<T> {
    //…
   return T;
}

@connect
export class Foo{

}

```

**Expected behavior:**
```ts
//in another file
import {Foo} from './Foo';

//this to work properly
const x = Foo.WrappedComponent;
```

**Work Around:**
```ts
class FooUnwrapped{

}
//call the annotation old-school
export const Foo = connect(FooUnwrapped);

```
```ts
//in another file
import {Foo} from './Foo';

//this to work properly
const x = Foo.WrappedComponent;
```

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/4881

**Comment by EisenbergEffect** [comment]
@mhegazy Is there any work going on to address this? It's a big deal for many frameworks now.  Any framework that uses HoCs, such as React, could use this. Aurelia vNext requires the same basic capability for its components as well.

**Comment by mhegazy** [comment]
We have been intentionally holding back on any decorator investments until we have implemented the new proposal. 
The new proposal seems to be stabilizing currently. we are just waiting on it to hit stage 3 in TC39, and then we will pick it up. once we have an implementation for the new proposal along with a migration path, we will be looking through our decorator feature request backlog and identifying possible updates. I am sure we will be sharing our plans once we have them.

**Comment by EisenbergEffect** [comment]
Please re-open as this is a critical issue for decorators when the work for the new proposal lands. @mhegazy

**Comment by mhegazy** [comment]
We have been using https://github.com/Microsoft/TypeScript/issues/4881 to track this request.

**Comment by EisenbergEffect** [comment]
Thank you!
